### PR TITLE
Fix Test-TargetResource and properties for IOS Wi-Fi Policy

### LINKED ISSUE #3973: IntuneWifiConfigurationPolicyIOS: Policy gets created but Test-DSCConfiguration fails with error
### Description of the issue

Same problem as described in #3970 and #3971, the policy gets created but the Id field is bogus so when running Test-DSCConfiguration it says that it found a policy with that Id (which isn't true) and Graph complains that it actually doesn't exist. Trying to deploy any changes to the policy result in same error message.

```powershell
VERBOSE: [REDACTED]:                            [[IntuneWifiConfigurationPolicyIOS]IntuneWifiConfigurationPolicyIOS-IntuneWifiConfigurationPolicyIOS_1] Testing configuration of
{8e809b9e-0032-40b7-b263-e6029daf8e9c}
VERBOSE: [REDACTED]:                            [[IntuneWifiConfigurationPolicyIOS]IntuneWifiConfigurationPolicyIOS-IntuneWifiConfigurationPolicyIOS_1] Found something with id
{8e809b9e-0032-40b7-b263-e6029daf8e9c}
[ResourceNotFound] : {
  "_version": 3,
  "Message": "DeviceConfigurations not found: Profile 8e809b9e-0032-40b7-b263-e6029daf8e9c - Operation ID (for customer support): 00000000-0000-0000-0000-000000000000 - Activity ID:
42528f46-f677-4930-8d87-c9b35d6074e0 - Url: https://fef.msub07.manage.microsoft.com/DeviceConfiguration_2311/StatelessDeviceConfigurationFEService/deviceManagement/deviceConfigurations('8e809b9e-0032-40b7-b263-
e6029daf8e9c')/assignments?api-version=5023-10-14",
  "CustomApiErrorPhrase": "",
  "RetryAfter": null,
  "ErrorSourceService": "",
  "HttpHeaders": "{}"
}
    + CategoryInfo          : InvalidOperation: ({ DeviceConfigu...ndProperty =  }:) [], CimException
    + FullyQualifiedErrorId : ResourceNotFound,Microsoft.Graph.Beta.PowerShell.Cmdlets.GetMgBetaDeviceManagementDeviceConfigurationAssignment_List
    + PSComputerName        : localhost
```

### Microsoft 365 DSC Version

1.23.1122.1

### Which workloads are affected

other

### The DSC configuration

```powershell
# Generated with Microsoft365DSC version 1.23.1122.1
# For additional information on how to use Microsoft365DSC, please visit https://aka.ms/M365DSC
param (
    [parameter()]
    [System.Management.Automation.PSCredential]
    $Credential
)

Configuration IntuneWifiConfigurationPolicyIOS
{
    param (
        [parameter()]
        [System.Management.Automation.PSCredential]
        $Credential
    )

    if ($null -eq $Credential)
    {
        <# Credentials #>
        $Credscredential = Get-Credential -Message "Credentials"

    }
    else
    {
        $CredsCredential = $Credential
    }

    $OrganizationName = $CredsCredential.UserName.Split('@')[1]

    Import-DscResource -ModuleName 'Microsoft365DSC' -ModuleVersion '1.23.1122.1'

    Node localhost
    {

        IntuneWifiConfigurationPolicyIOS "IntuneWifiConfigurationPolicyIOS-IntuneWifiConfigurationPolicyIOS_1"
        {
            Assignments                    = @(
                MSFT_DeviceManagementConfigurationPolicyAssignments{
                    dataType = '#microsoft.graph.groupAssignmentTarget'
                    deviceAndAppManagementAssignmentFilterType = 'none'
                    groupId = 'b0b8fd3f-af2a-453b-be57-80182d599f02'
                }
                MSFT_DeviceManagementConfigurationPolicyAssignments{
                    dataType = '#microsoft.graph.exclusionGroupAssignmentTarget'
                    deviceAndAppManagementAssignmentFilterType = 'none'
                    groupId = '053dc89a-be83-411a-bad3-909904b7239e'
                }
            );
            Credential                     = $CredsCredential;
            ConnectAutomatically           = $True;
            ConnectWhenNetworkNameIsHidden = $True;
            DisableMacAddressRandomization = $True;
            DisplayName                    = "IntuneWifiConfigurationPolicyIOS_1";
            Ensure                         = "Present";
            Id                             = "8e809b9e-0032-40b7-b263-e6029daf8e9c";
            NetworkName                    = "sf";
            ProxyAutomaticConfigurationUrl = "THSCP.local";
            ProxySettings                  = "automatic";
            Ssid                           = "sf";
            WiFiSecurityType               = "wpaPersonal";
        }
    }
}

IntuneWifiConfigurationPolicyIOS -ConfigurationData .\ConfigurationData.psd1 -Credential $Credential
```

### Verbose logs showing the problem

```powershell
VERBOSE: [REDACTED]:                            [[IntuneWifiConfigurationPolicyIOS]IntuneWifiConfigurationPolicyIOS-IntuneWifiConfigurationPolicyIOS_1] Testing configuration of
{8e809b9e-0032-40b7-b263-e6029daf8e9c}
VERBOSE: [REDACTED]:                            [[IntuneWifiConfigurationPolicyIOS]IntuneWifiConfigurationPolicyIOS-IntuneWifiConfigurationPolicyIOS_1] Found something with id
{8e809b9e-0032-40b7-b263-e6029daf8e9c}
[ResourceNotFound] : {
  "_version": 3,
  "Message": "DeviceConfigurations not found: Profile 8e809b9e-0032-40b7-b263-e6029daf8e9c - Operation ID (for customer support): 00000000-0000-0000-0000-000000000000 - Activity ID:
42528f46-f677-4930-8d87-c9b35d6074e0 - Url: https://fef.msub07.manage.microsoft.com/DeviceConfiguration_2311/StatelessDeviceConfigurationFEService/deviceManagement/deviceConfigurations('8e809b9e-0032-40b7-b263-
e6029daf8e9c')/assignments?api-version=5023-10-14",
  "CustomApiErrorPhrase": "",
  "RetryAfter": null,
  "ErrorSourceService": "",
  "HttpHeaders": "{}"
}
    + CategoryInfo          : InvalidOperation: ({ DeviceConfigu...ndProperty =  }:) [], CimException
    + FullyQualifiedErrorId : ResourceNotFound,Microsoft.Graph.Beta.PowerShell.Cmdlets.GetMgBetaDeviceManagementDeviceConfigurationAssignment_List
    + PSComputerName        : localhost
```

### Environment Information + PowerShell Version

```powershell
OsName               : Microsoft Windows 11 Enterprise
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 22621.1.amd64fre.ni_release.220506-1250
OsLanguage           : en-US
OsMuiLanguages       : {en-US, pt-PT}

Name                           Value
----                           -----
PSVersion                      5.1.22621.2428
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.22621.2428
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### PR DESCRIPTION
**Pull request by FabienTschanz**
#### Pull Request (PR) description
- Fixes an issue where the IOS Wi-Fi Profiles are not properly checked and verified by the DSC resource. 
- Updated the available properties

#### This Pull Request (PR) fixes the following issues
- Fixes #3973

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWifiConfigurationPolicyIOS/MSFT_IntuneWifiConfigurationPolicyIOS.psm1 [hunk 1]
```diff
@@ -257,7 +266,7 @@ function Set-TargetResource
         $Ssid,
 
         [Parameter()]
-        [ValidateSet('open', 'wpaPersonal', 'wpaEnterprise', 'wep', 'wpa2Personal', 'wpa2Enterprise')]
+        [ValidateSet('open', 'wpaPersonal', 'wep')]
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWifiConfigurationPolicyIOS/MSFT_IntuneWifiConfigurationPolicyIOS.psm1:269]: Removing values from the ValidateSet would be considered as breaking change. Any chance to keep the previous elements and wirte a warning about them being deprecated?

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWifiConfigurationPolicyIOS/MSFT_IntuneWifiConfigurationPolicyIOS.psm1 [hunk 2]
```diff
@@ -495,7 +523,7 @@ function Test-TargetResource
...
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWifiConfigurationPolicyIOS/MSFT_IntuneWifiConfigurationPolicyIOS.psm1:526]: Removing values from the ValidateSet would be considered as breaking change. Any chance to keep the previous elements and wirte a warning about them being deprecated?

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWifiConfigurationPolicyIOS/MSFT_IntuneWifiConfigurationPolicyIOS.schema.mof [hunk 3]
```diff
@@ -18,14 +18,15 @@ class MSFT_IntuneWifiConfigurationPolicyIOS : OMI_BaseResource
...
+    [Write, Description("If the pre shared key should be updated, even if the policy is already equal.")] Boolean ForcePreSharedKeyUpdate;
     [Write, Description("Network name")] String NetworkName;
     [Write, Description("Pre shared key")] String PreSharedKey;
     [Write, Description("Proxy automatic configuration url")] String ProxyAutomaticConfigurationUrl;
     [Write, Description("Proxy manual address")] String ProxyManualAddress;
     [Write, Description("Proxy manual port")] UInt32 ProxyManualPort;
     [Write, Description("Proxy settings"), ValueMap{"none","manual","automatic"}, Values{"none","manual","automatic"}] String ProxySettings;
     [Write, Description("SSID")] String Ssid;
-    [Write, Description("Wi-Fi security"), ValueMap{"open","wpaPersonal","wpaEnterprise","wep","wpa2Personal","wpa2Enterprise"}, Values{"open","wpaPersonal","wpaEnterprise","wep","wpa2Personal","wpa2Enterprise"}] String WiFiSecurityType;
+    [Write, Description("Wi-Fi security"), ValueMap{"open","wpaPersonal","wep"}, Values{"open","wpaPersonal","wep"}] String WiFiSecurityType;
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWifiConfigurationPolicyIOS/MSFT_IntuneWifiConfigurationPolicyIOS.schema.mof:29]: Removing values from the ValidateSet would be considered as breaking change. Any chance to keep the previous elements and wirte a warning about them being deprecated?

**Review by andikrueger — CHANGES_REQUESTED** [review]
Could you try to navigate around the breaking change for this fix?

**Comment by FabienTschanz** [comment]
@andikrueger It‘s not possible to create or export policies of the used odata type with the removed values in the ValidateSet. Intune does not allow the creation. 

Thus, I consider this not a breaking change, since the combination of e.g. a _wpa2Personal_ WiFi security type is not possible with the Odata type _iosWiFiConfiguration_. The Odata type for _wpa2Personal_ and enterprise is _iosEnterpriseWiFiConfiguration_.

**Comment by andikrueger** [comment]
Understood. Still they were present in the validate set and would break any configuration that might have had something that did not work implemented. Basically it's about compiling existing configurations. This change will break existing configurations due to the changed values.

Is this PR related to https://github.com/microsoft/Microsoft365DSC/pull/3989?

**Comment by FabienTschanz** [comment]
@andikrueger The PR is related but not dependent. All the Android Wi-Fi PRs aim to address the issues and clarify with the many available options.   
Any configuration that would have compiled with the now removed values in the `ValidateSet` could never have been deployed. Is really the possibility to compile the configuration the "goal", and not if the compiled objects can be deployed? If a configuration can be compiled but not deployed (since forever), that wouldn't make any sense to still make this a breaking change in my opinion. But if requested, I can create a new PR with only the necessary changes and introduce warnings for the deprecation.

**Pull request closed without merging**